src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/ReactorAccessPort.java

```diff
@@ -1,192 +1,233 @@
-		Inventory inv = inject(l);
-		Map<Integer, ItemStack> map = inv.addItem(item);
-		
-		for (int slot: getOutputSlots()) {
-			BlockStorage.getInventory(l).replaceExistingItem(slot, inv.getItem(slot));
-		}
-		
-		for (Map.Entry<Integer, ItemStack> entry: map.entrySet()) {
-			return entry.getValue();
-		}
-		return null;
-	}
-
-}
+package me.mrCookieSlime.Slimefun.Objects.SlimefunItem.machines;
+
+import java.util.Map;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import me.mrCookieSlime.CSCoreLibPlugin.CSCoreLib;
+import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.Item.CustomItem;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunBlockHandler;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.UnregisterReason;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AReactor;
+import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenuPreset;
+import me.mrCookieSlime.Slimefun.api.item_transport.ItemTransportFlow;
+
+public class ReactorAccessPort extends SlimefunItem {
+	
+	private static final int[] border = {0, 1, 2, 3, 4, 5, 6, 7, 8, 12, 13, 14, 21, 23};
+	private static final int[] border_1 = {9, 10, 11, 18, 20, 27, 29, 36, 38, 45, 46, 47};
+	private static final int[] border_2 = {15, 16, 17, 24, 26, 33, 35, 42, 44, 51, 52, 53};
+	private static final int[] border_3 = {30, 31, 32, 39, 41, 48, 49, 50};
+
+	public ReactorAccessPort(Category category, ItemStack item, String name, RecipeType recipeType, ItemStack[] recipe) {
+		super(category, item, name, recipeType, recipe);
+		
+		new BlockMenuPreset(name, getInventoryTitle()) {
+			
+			@Override
+			public void init() {
+				constructMenu(this);
+			}
+
+			@Override
+			public void newInstance(BlockMenu menu, Block b) {
+			}
+
+			@Override
+			public boolean canOpen(Block b, Player p) {
+				AReactor reactor = getReactor(b.getLocation());
+				if(reactor != null) {
+					boolean empty = true;
+					BlockMenu bm = getReactorMenu(b.getLocation());
+					if(bm != null) {
+						for(int slot:reactor.getCoolantSlots())
+							if(bm.getItemInSlot(slot) != null)
+								empty = false;
+						for(int slot:reactor.getFuelSlots())
+							if(bm.getItemInSlot(slot) != null)
+								empty = false;
+					
+						if(!p.isSneaking() && !empty && (p.hasPermission("slimefun.inventory.bypass") || CSCoreLib.getLib().getProtectionManager().canAccessChest(p.getUniqueId(),b,true))) {

```

Comment:
So Sneaking = Open the actual Inventory, Not sneaking = Open the Reactor Menu?
I think this should be swapped around.

Reply:
Well the reactor port almost always is empty as the items are quickly moved from the port to the reactor.    The reactor is the part that blows up if not taken care of properly so this is the reason for that functionality,       Since its the "REACTOR" access port I feel like it makes the most sense to open the Reactor's inventory first as that is the whole reason for the port's existence.